Frontend/library/src/Inputs/MouseControllerHovering.ts

```diff
@@ -0,0 +1,115 @@
+        this.onMouseWheelListener = this.onMouseWheel.bind(this);
+        this.onMouseMoveListener = this.onMouseMove.bind(this);
+        this.onContextMenuListener = this.onContextMenu.bind(this);
+    }
+
+    registerMouseEvents(): void {
+        super.registerMouseEvents();
+
+        this.videoElementParent.addEventListener('mousemove', this.onMouseMoveListener);
+        this.videoElementParent.addEventListener('mousedown', this.onMouseDownListener);
+        this.videoElementParent.addEventListener('mouseup', this.onMouseUpListener);
+        this.videoElementParent.addEventListener('contextmenu', this.onContextMenuListener);
+        this.videoElementParent.addEventListener('wheel', this.onMouseWheelListener);
+        this.videoElementParent.addEventListener('dblclick', this.onMouseDblClickListener);
+    }
+
+    unregisterMouseEvents(): void {
+        this.videoElementParent.removeEventListener('mousemove', this.onMouseMoveListener);
+        this.videoElementParent.removeEventListener('mousedown', this.onMouseDownListener);
+        this.videoElementParent.removeEventListener('mouseup', this.onMouseUpListener);
+        this.videoElementParent.removeEventListener('contextmenu', this.onContextMenuListener);
+        this.videoElementParent.removeEventListener('wheel', this.onMouseWheelListener);
+        this.videoElementParent.removeEventListener('dblclick', this.onMouseDblClickListener);
+
+        super.unregisterMouseEvents();
+    }
+
+    private onMouseDown(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
+            return;
+        }
+        const coord = this.coordinateConverter.normalizeAndQuantizeUnsigned(event.offsetX, event.offsetY);
+        this.streamMessageController.toStreamerHandlers.get('MouseDown')([event.button, coord.x, coord.y]);
+        event.preventDefault();
+    }
+
+    private onMouseUp(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
+            return;
+        }
+        const coord = this.coordinateConverter.normalizeAndQuantizeUnsigned(event.offsetX, event.offsetY);
+        this.streamMessageController.toStreamerHandlers.get('MouseUp')([event.button, coord.x, coord.y]);
+        event.preventDefault();
+    }
+
+    private onContextMenu(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
+            return;
+        }
+        event.preventDefault();
+    }
+
+    private onMouseMove(event: MouseEvent) {
+        if (!this.videoPlayer.isVideoReady()) {
+            return;
+        }
+        const coord = this.coordinateConverter.normalizeAndQuantizeUnsigned(event.offsetX, event.offsetY);
+        const delta = this.coordinateConverter.normalizeAndQuantizeSigned(event.movementX, event.movementY);
+        this.streamMessageController.toStreamerHandlers.get('MouseMove')([
+            coord.x,
+            coord.y,
+            delta.x,
+            delta.y
+        ]);
+        event.preventDefault();
```

Comment:
Same here about preventDefault()

Reply:
I can remove them but I didn't want to alter the existing behaviour too much.

Reply:
Let's keep it for now.